src/main/java/com/wynntils/modules/map/overlays/ui/MainWorldMapUI.java

```diff
@@ -134,14 +140,26 @@ private static boolean isHoldingMapKey() {
     @Override
     public void drawScreen(int mouseX, int mouseY, float partialTicks) {
         // HeyZeer0: This detects if the user is holding the map key;
-        if (!holdingMapKey && (System.currentTimeMillis() - creationTime >= 150) && isHoldingMapKey()) holdingMapKey = true;
-
+        if (!holdingDecided && (System.currentTimeMillis() - creationTime >= 150)) {
+            holdingDecided = true;
+            if (isHoldingMapKey())
+                holdingMapKey = true;
+        }
         // HeyZeer0: This close the map if the user was pressing the map key and after a moment dropped it
         if (holdingMapKey && !isHoldingMapKey()) {
             McIf.mc().displayGuiScreen(null);
             return;
         }
 
+        if (MapConfig.WorldMap.INSTANCE.autoCloseMapOnMovement)
+        {
+            if (checkForPlayerMovement(holdingDecided, holdingMapKey))
+            {
+                McIf.mc().displayGuiScreen(null);
+                return;
+            }
+        }
+
```

Comment:
Please follow our code standards!
```suggestion
        if (MapConfig.WorldMap.INSTANCE.autoCloseMapOnMovement && checkForPlayerMovement(holdingDecided, holdingMapKey))  {
            McIf.mc().displayGuiScreen(null);
            return;
        }

```